phoenix-core/src/it/java/org/apache/phoenix/end2end/MaxLookbackIT.java

```diff
@@ -0,0 +1,351 @@
+import static org.apache.phoenix.util.TestUtil.assertRawRowCount;
+import static org.apache.phoenix.util.TestUtil.assertRowExistsAtSCN;
+import static org.apache.phoenix.util.TestUtil.assertRowHasExpectedValueAtSCN;
+import static org.apache.phoenix.util.TestUtil.assertTableHasTtl;
+import static org.apache.phoenix.util.TestUtil.assertTableHasVersions;
+
+@NeedsOwnMiniClusterTest
+public class MaxLookbackIT extends BaseUniqueNamesOwnClusterIT {
+    private static final Log LOG = LogFactory.getLog(MaxLookbackIT.class);
+    private static final int MAX_LOOKBACK_AGE = 10;
+    private static final int ROWS_POPULATED = 2;
+    private String tableDDLOptions;
+    private StringBuilder optionBuilder;
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+        props.put(QueryServices.GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS_ATTRIB, Long.toString(0));
+        props.put(ScanInfoUtil.PHOENIX_MAX_LOOKBACK_AGE_CONF_KEY, Integer.toString(MAX_LOOKBACK_AGE));
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+    }
+
+    @Before
+    public void beforeTest(){
+        optionBuilder = new StringBuilder();
+        this.tableDDLOptions = optionBuilder.toString();
+    }
+
+    @Test
+    public void testTooLowSCNWithMaxLookbackAge() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String dataTableName = generateUniqueName();
+            String indexStem = generateUniqueName();
+            createTableAndIndexes(conn, dataTableName, indexStem);
+            //need to sleep long enough for the SCN to still find the syscat row for the table
+            Thread.sleep(MAX_LOOKBACK_AGE * 1000 + 1000);
+            Properties props = new Properties();
+            props.setProperty(PhoenixRuntime.CURRENT_SCN_ATTRIB,
+                Long.toString(EnvironmentEdgeManager.currentTime() - (MAX_LOOKBACK_AGE + 1) * 1000));
+            try (Connection connscn = DriverManager.getConnection(getUrl(), props)) {
+                connscn.createStatement().executeQuery("select * from " + dataTableName);
+            } catch (SQLException se) {
+                SQLExceptionCode code =
+                    SQLExceptionCode.CANNOT_QUERY_TABLE_WITH_SCN_OLDER_THAN_MAX_LOOKBACK_AGE;
+                TestUtil.assertSqlExceptionCode(code, se);
+                return;
+            }
+        }
+        Assert.fail("We should have thrown an exception for the too-early SCN");
+    }
+
+    @Test(timeout=120000L)
+    public void testRecentlyDeletedRowsNotCompactedAway() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String dataTableName = generateUniqueName();
+            String indexStem = generateUniqueName();
+            createTableAndIndexes(conn, dataTableName, indexStem);
```

Comment:
nit: You seem to be creating dataTable and indexes in each of these tests. Recommend you to move it to a @Before method

Reply:
Unfortunately the tables are slightly different each time.